.github/workflows/assign-class-label.yaml

```
@@ -0,0 +1,41 @@
+name: assign-class-label tests and linting/formatting
+
+on:
+  push:
+    paths:
+    - container-images/assign-class-label/**
+  pull_request:
+    paths:
+    - container-images/assign-class-label/**
+
+jobs:
+  lint-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'  
+
+      - name: Install dependencies
+        working-directory: ./container-images/assign-class-label/
+        run: |
+          pip install -r requirements.txt
+          pip install -r tests/requirements.txt
+
+      - name: Run ruff format (formatting)
+        run: |
+          ruff format
+
+      - name: Run ruff check (linting)
+        run: |
+          ruff check --fix
+
+      - name: Run tests
+        working-directory: ./container-images/assign-class-label/
+        run: |
+          PYTHONPATH=. pytest tests/
```

You should not need to manipulate `PYTHONPATH` in order to run your tests. Instead, create an empty file `container-images/assign-class-label/conftest.py`; this will allow pytest to locate your code.

```suggestion
          pytest tests/
```